torch/csrc/StorageMethods.cpp

```diff
@@ -173,8 +173,8 @@ static PyObject* THPStorage_fromBuffer(
   PyObject* dtype_obj = nullptr;
   c10::ScalarType scalar_type = at::kByte;
   Py_buffer buffer = {};
-  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,clang-diagnostic-writable-strings)
-  static char* kwlist[] = {
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
+  static const char* kwlist[] = {
```

Comment:
Set all char carrays in this PR to constexpr as this will enable compiletime bounds checking on supported compilers.